public/components/layer_config/documents_config/document_layer_style.tsx

```diff
@@ -17,49 +16,84 @@ import {
   EuiTitle,
   EuiFormRow,
   EuiForm,
+  useColorPickerState,
 } from '@elastic/eui';
 import { FormattedMessage } from '@osd/i18n/react';
 import { DocumentLayerSpecification } from '../../../model/mapLayerType';
 
 interface Props {
   selectedLayerConfig: DocumentLayerSpecification;
   setSelectedLayerConfig: Function;
+  setIsUpdateDisabled: Function;
 }
 
-export const DocumentLayerStyle = ({ setSelectedLayerConfig, selectedLayerConfig }: Props) => {
-  const [fillColor, setFillColor] = useState(selectedLayerConfig?.style?.fillColor);
-  const [borderColor, setBorderColor] = useState(selectedLayerConfig?.style?.borderColor);
+interface ColorPickerProps {
+  color: string;
+  setColor: (text: string, { isValid }: { isValid: boolean }) => void;
+  label: string;
+  colorError: (text: string, { isValid }: { isValid: boolean }) => void;
+}
+
+const ColorPicker = memo(({ color, setColor, label, colorError }: ColorPickerProps) => {
+  return (
+    <EuiFormRow label={label} fullWidth={true} isInvalid={!!colorError} error={colorError}>
+      <EuiColorPicker color={color} onChange={setColor} isInvalid={!!colorError} fullWidth={true} />
+    </EuiFormRow>
+  );
+});
+
+export const DocumentLayerStyle = ({
+  setSelectedLayerConfig,
+  selectedLayerConfig,
+  setIsUpdateDisabled,
+}: Props) => {
+  const [fillColor, setFillColor, fillColorErrors] = useColorPickerState(
+    selectedLayerConfig?.style?.fillColor
+  );
+  const [borderColor, setBorderColor, borderColorErrors] = useColorPickerState(
+    selectedLayerConfig?.style?.borderColor
+  );
   const [hasInvalidThickness, setHasInvalidThickness] = useState<boolean>(false);
   const [hasInvalidSize, setHasInvalidSize] = useState<boolean>(false);
   const geoTypeToggleButtonGroupPrefix = 'geoTypeToggleButtonGroup';
   const [toggleGeoTypeIdSelected, setToggleGeoTypeIdSelected] = useState(
     `${geoTypeToggleButtonGroupPrefix}__Point`
   );
 
+  // It's used to update the style color when switch layer config between different document layers
   useEffect(() => {
-    setFillColor(selectedLayerConfig?.style?.fillColor);
-    setBorderColor(selectedLayerConfig?.style?.borderColor);
-  }, [selectedLayerConfig]);
+    setFillColor(
+      selectedLayerConfig?.style?.fillColor,
+      !!fillColorErrors ? { isValid: false } : { isValid: true }
+    );
+    setBorderColor(
+      selectedLayerConfig?.style?.borderColor,
+      !!borderColorErrors ? { isValid: false } : { isValid: true }
+    );
+  }, [selectedLayerConfig.id]);
 
   useEffect(() => {
-    setSelectedLayerConfig({
-      ...selectedLayerConfig,
-      style: {
-        ...selectedLayerConfig?.style,
-        fillColor,
-      },
-    });
-  }, [fillColor]);
+    const disableUpdate =
+      !!fillColorErrors || !!borderColorErrors || hasInvalidSize || hasInvalidThickness;
+    setIsUpdateDisabled(disableUpdate);
+  }, [
+    setIsUpdateDisabled,
```

Comment:
do we have to add functions as depedency?

Reply:
Not have to, but It's a good practice to have all defined properties inside the component's scope so that use latest version of the function.